### PR TITLE
Clarify how setting a frontend priority works

### PR DESCRIPTION
**Pull request by jbdoumenjou**
### What does this PR do?

Update the documentation to clarify how setting a frontend priority works

### Motivation

Some misunderstanding on the frontend priority mechanism
Related to #2722 

### More

- [x] Added/updated documentations

### CODE REVIEW

File: docs/basics.md [hunk 1]
```diff
@@ -234,7 +234,7 @@ The following rules are both `Matchers` and `Modifiers`, so the `Matcher` portio
 #### Priorities
 
 By default, routes will be sorted (in descending order) using rules length (to avoid path overlap):
-`PathPrefix:/12345` will be matched before `PathPrefix:/1234` that will be matched before `PathPrefix:/1`.
+`PathPrefix:/12;Host:foo.com` (length == 27) will be matched before `PathPrefixStrip:/123` (length == 20) will be matched before `PathPrefix:/12,/3` (length == 17).
```
Comment (ldez) [inline docs/basics.md]: could you replace numbers by letters.

ex: `PathPrefix:/12` -> `PathPrefix:/foo`

File: docs/basics.md [hunk 2]
```diff
@@ -254,7 +254,7 @@ You can customize priority by frontend. The priority value is added to the rule
       rule = "PathPrefix:/toto"
 ```
 
-Here, `frontend1` will be matched before `frontend2` (`(3 + 10 == 13) > (4 + 5 == 9)`).
+Here, `frontend1` will be matched before `frontend2` (`(14 + 10 == 24) > (16 + 5 == 21)`).
```
Comment (Juliens) [inline docs/basics.md]: I think that in reality, if you specify a priority, you don't use the length and use directly the value. Can you verify this ?

Comment (timoreimann) [reply docs/basics.md]: In Traefik 1.5, a custom priority _does_ ignore the length-based factor according to my tests not too long ago.

Comment (jbdoumenjou) [reply docs/basics.md]: Yes, I made a mistake, the priority override the rule length according to https://github.com/containous/traefik/blob/master/server/server.go#L1192

**Pull request merged**